src/collective/exportimport/serializer.py

```diff
@@ -326,9 +326,10 @@ def __call__(self):
             return json_compatible(result)
 
     def get_at_blob_path(obj):
-        full_path = obj.getBlob().committed()
-        if not os.path.exists(full_path):
-            return
+        oid = obj._p_oid
```

Comment:
@pbauer Here you are getting the oid of the Archetypes object. In `get_dx_blob_path` you do `obj._blob._p_oid`. So there you get the oid of the blob. Shouldn't you do the same in the Archetypes case?
Note: I have not tried it, and don't have a website handy for quickly checking this.

Reply:
Ups, good catch, that's a error that I made during copy and paste. The `oid` needs to be from the blob and the `tid` needs to be from the object. 
I guess I should write a test for that feature...

Reply:
I tried it out in coredev 5.2 with Archetypes, creating both a dexterity and Archetypes file or image, and then trying out the code in `bin/instance-archetypes debug`. The current code actually seems okay, except for one thing, although I guess it works out in the end. In the Archetypes code we call `getBlobFilename`, which returns a full path. In the DX code we call `getBlobFilePath`, which returns the relative path from the blobstorage root. In both cases we then call `get_relative_blob_path` on the result, which gives us the relative path in both cases.
So looks like the code in this PR works, but could be simplified: call `getBlobFilePath` in both cases, and do not call `get_relative_blob_path`.

Reply:
Darn, I forgot that DemoStorage has no filesystem-blobs. I manually tested that it works though and added tests for the other two options...

Reply:
Thanks for the tip with `getBlobFilePath`!